core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -324,7 +327,10 @@ public void close() throws IOException {
         authSession.close();
       }
     } finally {
-      httpClient.close(CloseMode.GRACEFUL);
+      if (root) {
+        // Only close the underlying Apache HTTP client if this is a root HTTPClient.
+        httpClient.close(CloseMode.GRACEFUL);
```

Comment:
thanks for fixing this. I've noticed this during review and thought closing the parent instance is implicitly prevented, but I guess I missed this `finally` block

Reply:
I went ahead and implemented your suggestion.